alibi_detect/utils/missing_optional_dependency.py

```diff
@@ -0,0 +1,114 @@
+"""Functionality for optional importing
+This module provides a way to import optional dependencies. In the case that the user imports some functionality from
+alibi that is not usable due to missing optional dependencies this code is used to allow the import but replace it
```

Comment:
Change `alibi` to `alibi_detect`? Maybe worth a quick grep to check if any other places where comments have been copied from the `alibi` implementation?